dashboard/app/controllers/sections_controller.rb

```diff
@@ -1,10 +1,20 @@
 class SectionsController < ApplicationController
-  before_action :load_section
+  before_action :load_section_by_code, except: :update
 
   def show
     @secret_pictures = SecretPicture.all.shuffle
   end
 
+  # Allows you to update a section's course_id. Clears any assigned script_id
+  # in the process
+  def update
```

Comment:
`course_id` and `script_id` are very associated concepts here. Right now, only one or the other can be set (which is why updating course_id clears script_id). In the future, you'll be able to have both set, and the logic here will instead by something like:
```
if Course.find(params[:course_id]).scripts.map(&:id).contains(script_id) do
  # leave script_id alone
else
  # clear script_id
end